itests/ddo_test.go

```diff
@@ -0,0 +1,170 @@
+		Params: params,
+		Value:  big.Zero(),
+	}
+
+	sm, err := f.FullNode.MpoolPushMessage(ctx, msg, nil)
+	require.NoError(t, err, "AddVerifier failed")
+	res, err := f.FullNode.StateWaitMsg(ctx, sm.Cid(), 1, lapi.LookbackNoLimit, true)
+	require.NoError(t, err)
+	require.EqualValues(t, 0, res.Receipt.ExitCode)
+
+	// assign datacap to client
+	fileSize := 2000000
+	pieceSize := int64(8388608)
+	datacap := big.NewInt(pieceSize)
+
+	params, err = actors.SerializeParams(&verifregtypes.AddVerifiedClientParams{Address: f.ClientAddr, Allowance: datacap})
+	require.NoError(t, err)
+
+	msg = &types.Message{
+		From:   verifierAddr,
+		To:     verifreg.Address,
+		Method: verifreg.Methods.AddVerifiedClient,
+		Params: params,
+		Value:  big.Zero(),
+	}
+
+	sm, err = f.FullNode.MpoolPushMessage(ctx, msg, nil)
+	require.NoError(t, err)
+
+	res, err = f.FullNode.StateWaitMsg(ctx, sm.Cid(), 1, lapi.LookbackNoLimit, true)
+	require.NoError(t, err)
+	require.EqualValues(t, 0, res.Receipt.ExitCode)
+
+	// check datacap balance
+	dcap, err := f.FullNode.StateVerifiedClientStatus(ctx, f.ClientAddr, types.EmptyTSK)
+	require.NoError(t, err)
+	require.Equal(t, *dcap, datacap)
+
+	err = f.AddClientProviderBalance(abi.NewTokenAmount(1e15))
+	require.NoError(t, err)
+
+	// Create a CAR file
+	tempdir := t.TempDir()
+	randomFilepath, err := testutil.CreateRandomFile(tempdir, 5, fileSize)
+	require.NoError(t, err)
+	_, carFilepath, err := testutil.CreateDenseCARv2(tempdir, randomFilepath)
+	require.NoError(t, err)
+	commp, err := storagemarket.GenerateCommPLocally(carFilepath)
+	require.NoError(t, err)
+
+	pieceInfos := []string{fmt.Sprintf("%s=%d", commp.PieceCID, pieceSize)}
+	minerIds := []string{f.MinerAddr.String()}
+	allocateMsg, err := util.CreateAllocationMsg(ctx, f.FullNode, pieceInfos, minerIds, f.ClientAddr,
+		verifregst.MinimumVerifiedAllocationTerm,
+		verifregst.MaximumVerifiedAllocationTerm,
+		verifregst.MaximumVerifiedAllocationExpiration)
+	require.NoError(t, err)
+
+	sm, err = f.FullNode.MpoolPushMessage(ctx, allocateMsg, nil)
+	require.NoError(t, err)
+
+	_, err = f.FullNode.StateWaitMsg(ctx, sm.Cid(), 1, 1e10, true)
+	require.NoError(t, err)
+
+	allocations, err := f.FullNode.StateGetAllocations(ctx, f.ClientAddr, types.EmptyTSK)
+	require.NoError(t, err)
+	require.Len(t, allocations, 1)
+
+	var allocationId uint64
+	for id := range allocations {
+		allocationId = uint64(id)
+	}
+
+	dealUuid := uuid.New()
+	ddParams := smtypes.DirectDealParams{
+		DealUUID:           dealUuid,
+		AllocationID:       verifreg.AllocationId(allocationId),
+		PieceCid:           commp.PieceCID,
+		ClientAddr:         f.ClientAddr,
+		StartEpoch:         35000,
+		EndEpoch:           518400,
+		FilePath:           carFilepath,
+		DeleteAfterImport:  false,
+		RemoveUnsealedCopy: false,
+		SkipIPNIAnnounce:   false,
+	}
+	rej, err := f.Boost.BoostDirectDeal(ctx, ddParams)
+	require.NoError(t, err)
+	if rej != nil && rej.Reason != "" {
```

Comment:
I think we should be able to check for `rej.Reason == "Accepted"` even though the deal is not successfully sealed in a sector at the moment.

Reply:
If `Reason` is "" that means it was accepted - the `Reason` field is only filled in if there's an error (it's used to tell the client the reason why the deal was rejected, eg not enough datacap)